Utils: Diff against commit's ancestor

nonifier opened the pull request:
Make sure we diff the current branch with its forking point. 
This avoids finding differences in files that havent't been touched when the forking point isn't up to date with `origin/dev`

nonifier committed:
diff_against_ancestor

buildbotctp commented:
## Warning 
 You've modified files that potentially affect the model.     If so, for compatilility reason, please make sure you've updated `Data::data_version` in `source/type/data.cpp`. 

 Files: 
- `source/type/message.h`

nonifier committed:
Update check_retrocompat.sh

Diff against the first ancestor (aka the origin/dev merged into the current branch)
